### PR TITLE
add slice support for dim < 0

### PR DESCRIPTION
**Pull request by wopeizl**
test=develop

### CODE REVIEW

File: python/paddle/fluid/framework.py [hunk 1]
```diff
@@ -789,13 +789,24 @@ def __getitem__(self, item):
...
             newitem = self._reconstructSliceinfo(item) or item
-            check, info = self._detectContinuesSlice(newitem)
-            if check:
-                starts = info[0]
-                ends = info[1]
-                axes = [i for i in range(len(starts))]
-                return self._sliceVar(axes, starts, ends)
+            if fixedSize:
+                check, info = self._detectContinuesSlice(newitem)
+                if check and fixedSize:
```
Comment (junjun315) [inline python/paddle/fluid/framework.py:801]: ` and fixedSize` no need？

**Review by junjun315 — COMMENTED** [review]
a suggestion

File: python/paddle/fluid/framework.py (hunk 1 above)
Comment (wopeizl) [reply python/paddle/fluid/framework.py:801]: ok

**Review by junjun315 — APPROVED** [review]
fix it in next pr

**Pull request merged**